core/src/main/java/ar/com/wolox/wolmo/core/util/ImageUtils.java

```diff
@@ -0,0 +1,220 @@
+     * <p>
+     * <p/>
+     * Override the onActivityResult method in your fragment and specify behaviour
+     * for the provided request code. The selected image URI will be
+     * returned in the data variable of the activity result.
+     *
+     * @param fragment A fragment where the get image intent is going to be called
+     * @author juanignaciomolina
+     */
+    public static void getImageFromGallery(Fragment fragment, int requestCode, @StringRes int errorResId) {
+        Intent i = new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.INTERNAL_CONTENT_URI);
+        i.putExtra(Intent.EXTRA_LOCAL_ONLY, true);
+
+        // Ensure that there's a gallery app to handle the intent
+        if (i.resolveActivity(ContextUtils.getAppContext().getPackageManager()) != null)
+            fragment.startActivityForResult(i, requestCode);
+        else {
+            ToastUtils.showToast(errorResId);
+        }
+    }
+
+    /**
+     * Adds a given picture to the device images gallery
+     *
+     * @param imageUri The {@link Uri} of the image
+     */
+    public static void addPictureToDeviceGallery(Uri imageUri) {
+        Intent mediaScanIntent = new Intent(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE);
+        mediaScanIntent.setData(imageUri);
+        ContextUtils.getAppContext().sendBroadcast(mediaScanIntent);
+    }
+
+    /**
+     * Triggers an intent to go to the device's camera app, stores the image as 'filename'.'format',
+     * and returns its {@link Uri}
+     * <p>
+     * <p/>
+     * Override the onActivityResult method in your fragment and specify behaviour
+     * for the provided request code. This method returns a File object that
+     * contains the picture taken. You can get the returned image as an Uri using
+     * Uri.fromFile(getImageFromCamera(fragment));
+     *
+     * @param fragment    A fragment where the get image intent is going to be called
+     * @param requestCode A request code for the intent
+     * @param filename    Filename for the future stored image
+     * @param format      Format (extension) for the future image
+     * @param errorResId  Resource id of the error string
+     * @return {@link Uri} of the newly stored image
+     */
+    @Nullable
+    public static Uri getImageFromCamera(Fragment fragment, int requestCode, String filename,
+                                         @ImageFormat String format, @StringRes int errorResId) {
+
+        Intent i = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+
+        // Ensure that there's a camera app to handle the intent
+        if (i.resolveActivity(ContextUtils.getAppContext().getPackageManager()) != null) {
+            File photoFile = null;
+
+            try {
+                photoFile = FileUtils.createFile(filename, format);
+            } catch (IOException ex) {
+                ToastUtils.showToast(errorResId);
+            }
+
+            Uri photoFileUri = Uri.fromFile(photoFile);
```

Comment:
NPE Warning! If the file could not be created (IOException is thrown), `photoFile` will be `null` and `Uri.fromFile(photoFile)` will throw a [NPE](https://developer.android.com/reference/android/net/Uri.html#fromFile(java.io.File)).

I think this line should be inside the `if` below.